internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,199 @@
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+	containerName    = "container-00"
+)
+
+//nolint:funlen
+func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
+	dsAnnotations := make(map[string]string)
+	dsAnnotations["debug.openshift.io/source-container"] = containerName
+	dsAnnotations["openshift.io/scc"] = nodeExporter
+	matchLabels := make(map[string]string)
+	matchLabels["name"] = dsName
+
+	var runAsPrivileged = true
+	var zeroInt int64
+	var zeroInt32 int32
+	var preempt = corev1.PreemptLowerPriority
+	var tolerationsSeconds int64 = 300
+	var hostPathType = corev1.HostPathDirectory
+
+	container := corev1.Container{
+		Name:            containerName,
+		Image:           imageWithVersion,
+		ImagePullPolicy: "IfNotPresent",
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &runAsPrivileged,
+			RunAsUser:  &zeroInt,
+		},
+		Stdin:                  true,
+		StdinOnce:              true,
+		TerminationMessagePath: "/dev/termination-log",
+		TTY:                    true,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: "/host",
+				Name:      "host",
+			},
+		},
+	}
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       []corev1.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+					HostNetwork:      true,
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	tnf.ClaimFilePrintf("Deleting daemon set %s\n", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	if err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
```

Comment:
Instead of breaking this `if err ...` into multiple lines. Why not just a simple assignment and a following check?
```
	err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
		PropagationPolicy: &deletePolicy,})
	if err != nil {
		tnf.ClaimFilePrintf("The daemonset (%s) deletion is unsuccessful due to %s", daemonSetName, err)
	}
```